src/language-js/needs-parens.js

```diff
@@ -456,7 +456,14 @@ function needsParens(path, options) {
         (parent.type === "TSTypeAnnotation" &&
           path.getParentNode(1).type.startsWith("TSJSDoc"))
       );
-
+    case "TSTypeQuery":
+      if (
+        parent.type === "TSIndexedAccessType" ||
+        parent.type === "TSArrayType"
+      ) {
+        return true;
+      }
+    // fallthrough
```

Comment:
I don't think `TSTypeQuery`(TypeScript node type) can be a child of `NullableTypeAnnotation`(flow node type). This should not fallthrough.